software/react-native/concepts.md

```diff
@@ -0,0 +1,60 @@
+
+2. A saga is listens for this `attempting` action, and then runs when it receives it along with it's params (*it will have passed through the reducer first*). The saga handles all side-effects associated with this action. The saga will be the place where the api call is actually made. Depending on it's success or failure a `success` or `failure` action will be dispatched to the reducer. If it's a `success` we dispatch the received data as well, if it's a `failure`, we display the received error messages.
+
+3. After the `success` action is dispatched from the reducer, the data will get stored as is or [normalized](https://github.com/paularmstrong/normalizr) and stored. We try to normalize all deeply nested data to make it much easier to work with. The `success` or `failure` action will also flip that `attempting: true` flag mentioned above back to `attempting: false` to show the user that the request has completed.
+
+4. Now that we have attempted to fetch our data, successfully received our data, and stored (or normalized) it, depending on how we are using it, we may need to *transform* the data before displaying or performing calculations with it.
+In a perfect world, we would not need to do many transformations since the API would return to us the data exactly as we needed *cough GraphQL*. However, often transforms are necessary to coerce the data into a more workable form. For example, List Views in React Native require very specific data structures, and we often have to transform our data to fit a List View's requirements.
+
+5. The cycle is now complete. To recap: we first dispatch a redux action which may trigger an attempting flag. The saga that is listening for this action receives it and any params passed in and makes the actual API call. Regardless of success or failure, the saga will dispatch an action back to the reducer with the result of the API call. If it was a success we may normalize the data before we store it. If it was a failure, we may display the error messages. Finally, depending on how we want to use the data, we may transform it before using it in the UI.
+
+
+
+#### Redux Saga
+
+[Redux Sagas](https://github.com/yelouafi/redux-saga) are middleware for handling side-effects in Redux.
+
+##### Handling API calls with Sagas
+
+All API calls are handled by Sagas.
+
+There are generally three parts to handling an api call in react-native.
+
+1. The first phase is the request **attempt** itself.
+2. The second is to handle the **success** of the call with the successful payload response.
+3. The third is to handle the **failure** of the call, with an error payload response.
+
+All three of these cases involve separate state changes.
+
+In order to handle these three aspects, each API call generally has three different `action creators` associated with it.
```

Comment:
should expand on `action creators` a bit, maybe link to documentation about what they are, but also talk about the fact that typically we have a constants file that we just throw in so that we don't end up mis-typing strings